compiler/luci/pass/src/ResolveCustomOpMaxPoolWithArgmaxPass.cpp

```diff
@@ -0,0 +1,668 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/ResolveCustomOpMaxPoolWithArgmaxPass.h"
+
+#include "flatbuffers/flexbuffers.h"
+#include <loco/IR/DataTypeTraits.h>
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+#include <loco.h>
+#include <oops/InternalExn.h>
+
+namespace
+{
+
+template <typename T> std::vector<T> to_vector(const flexbuffers::TypedVector &typed_vec)
+{
+  std::vector<T> answer(typed_vec.size());
+
+  for (uint32_t i = 0; i < answer.size(); ++i)
+  {
+    answer[i] = typed_vec[i].As<T>();
+  }
+
+  return answer;
+}
+
+luci::Padding string_to_padding(const std::string &pad_str)
+{
+  if (pad_str == "VALID")
+    return luci::Padding::VALID;
+  if (pad_str == "SAME")
+    return luci::Padding::SAME;
+
+  return luci::Padding::UNDEFINED;
+}
+
+template <typename NodeT> void set_stride(NodeT *node, const luci::Stride &stride)
+{
+  node->stride()->h(stride.h());
+  node->stride()->w(stride.w());
+}
+
+template <typename NodeT> void set_filter(NodeT *node, const luci::Filter &filter)
+{
+  node->filter()->h(filter.h());
+  node->filter()->w(filter.w());
+}
+
+void copy_shape(const luci::CircleNode *src, luci::CircleNode *dst)
+{
+  assert(src->shape_status() == luci::ShapeStatus::VALID);
+
+  dst->rank(src->rank());
+  for (uint32_t i = 0; i < dst->rank(); ++i)
+  {
+    dst->dim(i).set(src->dim(i).value());
+  }
+
+  dst->shape_status(luci::ShapeStatus::VALID);
+}
+
+void init_name_and_origin(luci::CircleNode *node, const std::string &name,
+                          const std::shared_ptr<luci::CircleNodeOrigin> &origin)
+{
+  node->name(name);
+  luci::add_origin(node, origin);
+}
+
+void set_shape_and_dtype_from_node(const luci::CircleNode *src, luci::CircleNode *dst)
+{
+  copy_shape(src, dst);
+  dst->dtype(src->dtype());
+}
+
+template <typename NodeT> NodeT *none_act_func(NodeT *node)
+{
+  node->fusedActivationFunction(luci::FusedActFunc::NONE);
+  return node;
+}
+
+template <loco::DataType DT> luci::CircleCast *create_cast(luci::CircleNode *input)
+{
+  auto cast = input->graph()->nodes()->create<luci::CircleCast>();
```

Comment:
If we use `CircleCast`, can we handle this at internal backend @.@?

Reply:
Good question. And `circle-quantizer` too...

Reply:
@llFreetimeII
> If we use CircleCast, can we handle this at internal backend @.@?

We can do it! (but with some restrictions =/ )

Currently I am working on it.